components/ApiKeyModal/index.tsx

```diff
@@ -60,50 +74,65 @@ const ApiKeyModal = ({ open }: any) => {
 
       return true
     },
-    [query?.kanmonApiKey],
+    [query?.kanmonApiKey, query?.useCdnSdk],
   )
 
   // Prevent flickering when api key is passed through query param
   if (!doneValidatedApiKeyFromQueryParam) return null
 
+  const initialValues: FormValues = {
+    apiKey: '',
+    useCdnSdk: false,
+  }
+
   return (
     <Modal open={open}>
       <>
         <div className="top-1/2 left-1/2 absolute transform -translate-x-1/2 -translate-y-1/2 bg-white text-center rounded w-[450px] md:w-[550px]">
           <div className="p-12">
-            <div className="mt-4">
-              <h1 className="text-xl font-semibold mb-8">
-                Access the Demo Platform with your API Key
-              </h1>
-              <form
-                onSubmit={(e) => {
-                  e.preventDefault()
-                  saveApiKeyFn(localApiKey)
-                }}
-              >
-                <div className="mb-8">
-                  <TextField
-                    variant="outlined"
-                    value={localApiKey}
-                    placeholder={'Set Api Key'}
-                    fullWidth
-                    onChange={(e) => {
-                      setLocalApiKey(e.target.value)
-                    }}
-                  />
-                </div>
-
-                <Button
-                  fullWidth
-                  variant="contained"
-                  disabled={!localApiKey || saveApiKeyLoading}
-                  color="primary"
-                  type="submit"
-                >
-                  Start new demo
-                </Button>
-              </form>
-            </div>
+            <Formik
+              onSubmit={async (values) => {
+                await saveApiKeyFn(values)
+              }}
+              initialValues={initialValues}
+              validationSchema={apiKeyValidationSchema}
+              validateOnMount={true}
+            >
+              {({ isValid, handleSubmit, isSubmitting }) => {
+                return (
+                  <div className="mt-4">
+                    <h1 className="text-xl font-semibold mb-8">
+                      Access the Demo Platform with your API Key
+                    </h1>
+                    <Form>
+                      <div className="mb-4">
+                        <FormikTextInput<FormValues>
+                          fieldName="apiKey"
+                          placeholder={'Set Api Key'}
+                        />
+                      </div>
+                      <div className="col-span-6">
+                        <FormikCheckboxInput<FormValues>
+                          fieldName="useCdnSdk"
+                          label="Use CDN SDK"
+                          containerClass="flex items-center mb-4"
+                        />
```

Comment:
I think we can improve the UX here a bit. Like a toggle state or switcher or something.

1. web-sdk NPM Package (recommended)
2. sdk via Kanmon CDN

Default should be option 1 selected, requiring someone to intentionally change to option 2 for testing reasons.